packages/web3-eth-ens/test/unit/registry.test.ts

```diff
@@ -0,0 +1,870 @@
+			expect(setResolverMock).toHaveBeenCalledWith(namehash(ENS_NAME), mockAddress);
+			expect(send).toHaveBeenCalledWith(sendOptions);
+		});
+		it('setResolver throw (transaction reverted)', async () => {
+			// eslint-disable-next-line @typescript-eslint/no-empty-function
+			const send = jest.spyOn({ send: () => {} }, 'send').mockImplementation(() => {
+				throw new Error();
+			});
+
+			const setResolverMock = jest
+				.spyOn(registry['contract'].methods, 'setResolver')
+				.mockReturnValue({
+					send,
+				} as unknown as NonPayableMethodObject<any, any>);
+
+			const sendOptions = { from: mockAddress };
+
+			await expect(async () => {
+				await registry.setResolver(ENS_NAME, mockAddress, sendOptions);
+			}).rejects.toThrow(new Error());
+			expect(setResolverMock).toHaveBeenCalledWith(namehash(ENS_NAME), mockAddress);
+			expect(send).toHaveBeenCalledWith(sendOptions);
+		});
+
+		it('resolver', async () => {
+			// eslint-disable-next-line @typescript-eslint/no-empty-function
+			const call = jest
+				.spyOn(
+					{
+						call: async () => {
+							return mockAddress;
+						},
+					},
+					'call',
+				)
+				.mockReturnValue(Promise.resolve(mockAddress));
+
+			const resolverMock = jest
+				.spyOn(registry['contract'].methods, 'resolver')
+				.mockReturnValue({
+					call,
+				} as unknown as NonPayableMethodObject<any, any>);
+
+			await registry.getResolver(ENS_NAME);
+			expect(resolverMock).toHaveBeenCalledWith(namehash(ENS_NAME));
+			expect(call).toHaveBeenCalled();
+		});
+
+		it('resolver (return non string)', async () => {
+			// eslint-disable-next-line @typescript-eslint/no-empty-function
+			const call = jest
+				.spyOn(
+					{
+						call: async () => {
+							return 5; // something that is not string
+						},
+					},
+					'call',
+				)
+				.mockReturnValue(Promise.resolve(5)); // something that is not string
+
+			const resolverMock = jest
+				.spyOn(registry['contract'].methods, 'resolver')
+				.mockReturnValue({
+					call,
+				} as unknown as NonPayableMethodObject<any, any>);
+
+			await expect(async () => {
+				await registry.getResolver(ENS_NAME);
+			}).rejects.toThrow(new Error());
+			expect(resolverMock).toHaveBeenCalledWith(namehash(ENS_NAME));
+			expect(call).toHaveBeenCalled();
+		});
+
+		it('resolver throw (transaction reverted)', async () => {
+			// eslint-disable-next-line @typescript-eslint/no-empty-function
+			const call = jest.spyOn({ call: () => {} }, 'call').mockImplementation(() => {
+				throw new Error();
+			});
+			const resolverMock = jest
+				.spyOn(registry['contract'].methods, 'resolver')
+				.mockReturnValue({
+					call,
+				} as unknown as NonPayableMethodObject<any, any>);
+
+			await expect(async () => {
+				await registry.getResolver(ENS_NAME);
+			}).rejects.toThrow(new Error());
+			expect(resolverMock).toHaveBeenCalledWith(namehash(ENS_NAME));
+			expect(call).toHaveBeenCalled();
+		});
+	});
+	// it('Doesn"t support interface', async () => {
```

Comment:
Why these tests are commented? you there is any thing pending `.skip` should be used else these should be remoevd?

Reply:
I copied tests from `resolver.test.ts` and worked on them. I forgot to delete them, thanks for the catch!